text/0000-terminology-and-boundaries.md

```diff
@@ -0,0 +1,224 @@
+- Feature Name: Terminology and Boundary Definition
+- Start Date: 2018-10-11
+- RFC PR: (leave this empty)
+- Nebulet Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The Nebulet operating system, being so different from popular operating systems
+of today, will require new terminology to refer to abstraction boundaries.
+Terms such as "userspace" and "kernelspace" will carry too much baggage from
+existing OSes, which is likely to confuse potential contributors. This RFC
+proposes we define the following terms to refer to parts of our OS, adapted
+from [EROS](https://web.archive.org/web/20160412201504/http://www.coyotos.org/docs/misc/eros-structure.pdf):
+the _kernel_, which implements threading and inter-proces communication (IPC)
+as a wasm module, the _nucleus_, which implements a small wasm JIT natively,
```

Comment:
Just to be clear, the kernel will be a wasm module that runs on the nucleus?

Reply:
Yes, this has been clarified in my latest revision.